src/task_scl_eff_pot_hab.py

```diff
@@ -206,38 +261,76 @@ def dilate(image, distance):
 
         scl_polys = potential_core.add(potential_stepping_stone).selfMask()
 
-        # Temp geometry for RFE scenario
-        # geometry = ee.Geometry.Polygon(
-        #     [
-        #         [
-        #             [129.28241561583914, 51.490498638741165],
-        #             [129.28241561583914, 42.20681310459823],
-        #             [140.35663436583914, 42.20681310459823],
-        #             [140.35663436583914, 51.490498638741165],
-        #         ]
-        #     ],
-        #     None,
-        #     False,
-        # )
+        # TODO: the probability threshold process will change with final version of probability
+        probability_thresh = (
+            self.probability.add(1).gte(self.thresholds["probability"]).selfMask()
+        )
+
+        # TODO: incorporate actual survey effort input
+        survey_effort = ee.Image.constant(1)
 
         scl_polys = (
             ee.Image(1)
             .updateMask(scl_polys)
             .addBands(scl_polys)
-            .addBands(self.probability)
-            .rename(["constant", "scl_poly", "probability"])
+            .addBands(probability_thresh)
+            .addBands(survey_effort)
+            .rename(["scl_poly", "size", "probability", "effort"])
             .reduceToVectors(
-                reducer=ee.Reducer.max(),
-                geometry=self.extent,
+                reducer=ee.Reducer.max(),  # TODO: need to consider reducer for each band to delineate polygons
+                geometry=ee.Geometry.MultiPolygon(self.aoi).bounds(),
                 scale=self.scale,
                 crs="EPSG:4326",
                 maxPixels=self.ee_max_pixels,
             )
         )
 
-        export_path = "scl_poly/{}/scl_polys".format(self.taskdate)
+        scl_species = scl_polys.filter(
+            ee.Filter.And(
+                ee.Filter.gte("size", self.scl_poly_filters["scl_species"]["size"]),
+                ee.Filter.gte(
+                    "probability", self.scl_poly_filters["scl_species"]["probability"]
+                ),
+                ee.Filter.gte("effort", self.scl_poly_filters["scl_species"]["effort"]),
+            )
+        )
+
+        scl_survey = scl_polys.filter(
+            ee.Filter.And(
+                ee.Filter.gte("size", self.scl_poly_filters["scl_survey"]["size"]),
+                ee.Filter.gte(
+                    "probability", self.scl_poly_filters["scl_survey"]["probability"]
+                ),
+                ee.Filter.lte("effort", self.scl_poly_filters["scl_survey"]["effort"]),
```

Comment:
With the current filters structure, there is a logical possibility of overlaps (e.g. here we're using `lte` and above the effort threshold is using `gte`; one of these should probably leave out the `e`). Wondering if a simple filter structure would make sense?